chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -767,128 +797,237 @@ impl<'a> Radio<'a> {
+                        self.rx_client.map(|client| {
+                            start_task = true;
+                            client.receive(
+                                rbuf,
+                                frame_len,
+                                self.registers.crcstatus.get() == 1,
+                                result,
+                            );
+                        });
+                    }
+                }
             }
+            RadioState::TX => {
+                // Confirm proper hardware shortcuts are enabled
+                if !self.registers.shorts.matches_all(
+                    Shortcut::DISABLED_RXEN::SET
+                        + Shortcut::CCAIDLE_TXEN::SET
+                        + RXREADY_CCASTART::SET,
+                ) {
+                    panic!("Incorrect shortcut configuration for TX")
+                }
 
-            self.enable_interrupts();
-        }
+                // Handle Event_ready interrupt. The TX path performs both a TX ramp up and
+                // an RX ramp up. This means that there are two potential cases we must handle.
+                // The ready event due to the Rx Ramp up shortcuts to the CCASTART while the ready
+                // event due to the Tx ramp up requires we issue a start task in response to progress
+                // the state machine.
+                if self.registers.event_ready.is_set(Event::READY) {
+                    // In both cases, we must clear event
+                    self.registers.event_ready.write(Event::READY::CLEAR);
+
+                    // Ready event from Tx ramp up will be in radio internal TXIDLE state
+                    if self.registers.state.get() == nrf5x::constants::RADIO_STATE_TXIDLE {
+                        start_task = true;
+                    }
+                }
 
-        // tx or rx finished!
-        if self.registers.event_end.is_set(Event::READY) {
-            self.registers.event_end.write(Event::READY::CLEAR);
+                // Handle CCA related interrupts.
+                if self.registers.event_ccabusy.is_set(Event::READY) {
+                    self.registers.event_ccabusy.write(Event::READY::CLEAR);
+
+                    //need to back off for a period of time outlined
+                    //in the IEEE 802.15.4 standard (see Figure 69 in
+                    //section 7.5.1.4 The CSMA-CA algorithm of the
+                    //standard).
+                    if self.cca_count.get() < IEEE802154_MAX_POLLING_ATTEMPTS {
+                        self.cca_count.set(self.cca_count.get() + 1);
+                        self.cca_be.set(self.cca_be.get() + 1);
+                        let backoff_periods = self.random_nonce() & ((1 << self.cca_be.get()) - 1);
+                        let current_time = self.timer0.unwrap_or_panic().get_current_time();
+                        self.timer0
+                            .unwrap_or_panic() // Unwrap fail = Missing timer reference for CSMA
```

Comment:
This was already here, so not critical to fix at this stage, but this really shouldn't be a runtime check / possible failure. This should be passed in with component creation, as it's not possible to implement this radio (correctly) without a timer. If you're already mucking with components, would be nice to eliminate this...